crates/lib3h/src/dht/mirror_dht.rs

```diff
@@ -119,21 +135,94 @@ impl Dht for MirrorDht {
                 error!("serve_DhtCommand() failed: {:?}", res);
             }
         }
+        // Check if others timed-out
+        let mut timed_out_list = Vec::new();
+        for (peer_address, peer) in self.peer_map.iter() {
+            // Skip self
+            if peer_address == &self.this_peer.peer_address {
+                continue;
+            }
+            // Skip already timed out
+            let has_timed_out = self
+                .timed_out_map
+                .get(peer_address)
+                .expect("Should allways have time_out value for a peer");
+            if *has_timed_out {
+                continue;
+            }
+            // Check if timed-out
+            if now - peer.timestamp > self.config.timeout_threshold {
+                debug!("@MirrorDht@ peer {} timed-out", peer_address);
+                outbox.push(DhtEvent::PeerTimedOut(peer_address.clone()));
+                timed_out_list.push(peer_address.clone());
+                did_work = true;
+            }
+        }
```

Comment:
Walking the peer map is a bit expensive. You could use a BTreeMap<u64, String> and invoke the [range](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.range) function over the time interval of interest. You just need to make sure to update the map every time a node communicates.

Another way I might have optimized this in a past life is by maintaining a low priority background thread which periodically does these checks outside of the critical process loop but I rather not introduce concurrency to our framework yet.

Reply:
yeah, and not worth the effort for mirrorDHT as its temp stuff until we get rrdht.